sign_verify_file_ssh.sh

```diff
@@ -0,0 +1,128 @@
+#!/bin/bash
+
+# Usage message
+usage() {
+    echo "Usage: $0 sign <file> <private_key> | verify <file> <public_key> [signature_file]"
+    exit 1
+}
+
+# Ensure at least two arguments are provided
+if [ "$#" -lt 3 ]; then
+    usage
+fi
+
+MODE="$1"
+FILE_TO_SIGN="$2"
+
+# Ensure the file exists
+if [ ! -f "$FILE_TO_SIGN" ]; then
+    echo "Error: File '$FILE_TO_SIGN' not found."
+    exit 1
+fi
+
+# Function to securely convert PEM to OpenSSH if needed
+convert_pem_to_ssh() {
+    local key_file="$1"
+    local output_file="$2"
+
+    if grep -q "BEGIN RSA PRIVATE KEY" "$key_file" || grep -q "BEGIN OPENSSH PRIVATE KEY" "$key_file"; then
+        # Ensure cleanup on exit
+        trap 'rm -f "$output_file" "$output_file.pub"' EXIT
+
+        # Convert PEM key to OpenSSH format
+        cp "$key_file" "$output_file" && ssh-keygen -c -C "Converted from PEM" -f "$output_file" || {
+            echo "Error: Failed to convert PEM key to OpenSSH."
+            exit 1
+        }
+
+        # Generate the public key
+        ssh-keygen -y -f "$key_file" > "$output_file.pub" || {
+            echo "Error: Failed to generate public key from PEM key."
+            exit 1
+        }
```

Comment:
See comment about unusual code style above (this one here is easier to read though).

What happens if the key is protected by a passphrase?

Reply:
So there's an implicit assumption here that it is a GitHub App key and therefore has no passphrase. I can add support for an environment variable if there is a passphrase. I'm sure there must be a way to query if the key needs a passphrase...

Reply:
Passphrase is now supported